snowfakery/plugins.py

```diff
@@ -102,6 +111,23 @@ def lazy(func: Any) -> Callable:
     return func
 
 
+def resolve_plugins(
+    plugin_specs: List[Tuple[str, object]], search_paths: List[Union[str, Path]]
+):
+    "Resolve a list of plugins and lineinfos"
+    cwd_plugins = "./plugins"
+    user_plugins = Path.home() / ".snowfakery/plugins"
+    new_sys_path = [
+        *sys.path,
+        *(str(p) for p in search_paths),
+        str(cwd_plugins),
+        str(user_plugins),
+    ]
+
+    with patch.object(sys, "path", new_sys_path):
```

Comment:
This looks potentially dangerous if snowfakery is ever embedded in a threaded context, but I'm not sure what to suggest as an alternative.